### DISCUSSION TITLE
Out of memory on Colab T4

### DISCUSSION THREAD

**Discussion by martinambrus** [yl4579/StyleTTS2#102] [Q&A] [answered]
I tried to set this project on Google Colab with a free account and a T4 grapics card. Everything works until I get to epoch 50 of first stage, which gives me an out of memory error.

I already have `batch_size` set to 4.

I was wondering what does the `max_len` setting actually mean or do in reality. The comment is that it's 400 frames. In issue [#10](https://github.com/yl4579/StyleTTS2/issues/10#issuecomment-1751785656), it's said that this is an equivalent to 5 seconds. Does it mean that the training will not take more than 5 seconds of each wav file? 

My question is what happens if I raise/lower that `max_len` setting? Would it affect the ouput too much if I take it down to let's say 3 seconds?

Thanks to anyone who can shed some light into this for me :)

**Comment by yl4579** [comment, marked as answer]
Please see https://github.com/yl4579/StyleTTS2/issues/52#issuecomment-1821762635 and #81
